libdnf/rpm/solv/package_private.cpp

```diff
@@ -69,5 +69,14 @@ const char * get_full_nevra(Pool * pool, libdnf::rpm::PackageId package_id) {
     return output_string;
 }
 
-}  // namespace libdnf::rpm::solv
+std::string get_local_filepath(Pool * pool, libdnf::rpm::PackageId package_id) {
+    auto solvable = get_solvable(pool, package_id);
+    if (auto repo = static_cast<Repo *>(solvable->repo->appdata)) {
+        auto dir = std::filesystem::path(repo->get_cachedir()) / "packages";
+        return dir / std::filesystem::path(get_location(pool, package_id)).filename();
+    } else {
+        return "";
+    }
+}
```

Comment:
What about local repositories? According to the original code in python, it uses baseurl+get_location(pool, package_id).

Reply:
According to our discussion, I added todo.